File: Amazon_LeetCode_Python_C++/ArrayString/3_longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = 0
        mp = dict()
        l = 0
        for r in range(len(s)):
            if s[r] in mp:
                l = max(l, mp[s[r]])
            ans = max(ans, r - l + 1)
            mp[s[r]] = r + 1
        
        return ans

File: LeetCode/Explore/TopInterview/Easy/Strings/Blind75/3_longest_substring_without_repeating_characters.cpp
class Solution {
public:
  int lengthOfLongestSubstring(string s) {
    int maxLen = 0;
    unordered_map<int, int> mp;
    for (int r = 0, l = 0; r < s.size(); r++) {
      if (mp.count(s[r])) {
        l = max(l, mp[s[r]]);
      }
      maxLen = max(maxLen, r - l + 1);
      mp[s[r]] = r + 1;
    }
    return maxLen;
  }
};
